backend/Recodo.API/Controllers/AuthController.cs

```diff
@@ -43,5 +40,26 @@ public async Task<ActionResult<AuthUserDTO>> Login([FromBody] LoginUserDTO userD
         {
             return Ok(await _authService.Authorize(userDTO));
         }
+
+        [HttpPost("GoogleLogin")]
+        public async Task<IActionResult> GoogleLogin([FromBody] ExternalAuthDto externalAuth)
+        {
+            var payload = await _authService.VerifyGoogleToken(externalAuth);
+            if (payload == null)
+            {
+                return BadRequest("Invalid External Authentication.");
+            }
+
```

Comment:
Can you check payload inside the VerifyGoogleToken method, and if it is not valid, throw out new Exception like VerifyGoogleTokenException that will be handled by ErrorHandlerMiddleware

Reply:
Done